ChallengesWithTestsMarkVII/ChallengesSet07.cs

```diff
@@ -0,0 +1,164 @@
+            }
+            else if (finalBoard[0, 0] == 'O' && finalBoard[0, 1] == 'O' && finalBoard[0, 2] == 'O')
+            {
+                return TicTacToeResult.O;
+            }
+            else if (finalBoard[0, 1] == 'O' && finalBoard[1, 1] == 'O' && finalBoard[2, 1] == 'O')
+            {
+                return TicTacToeResult.O;
+            }
+            else if (finalBoard[0, 2] == 'O' && finalBoard[1, 2] == 'O' && finalBoard[2, 2] == 'O')
+            {
+                return TicTacToeResult.O;
+            }
+            else if (finalBoard[0, 0] == 'O' && finalBoard[1, 1] == 'O' && finalBoard[2, 2] == 'O')
+            {
+                return TicTacToeResult.O;
+            }
+            else if (finalBoard[1, 0] == 'O' && finalBoard[1, 1] == 'O' && finalBoard[1, 2] == 'O')
+            {
+                return TicTacToeResult.O;
+            }
+            else if (finalBoard[2, 0] == 'O' && finalBoard[2, 1] == 'O' && finalBoard[2, 2] == 'O')
+            {
+                return TicTacToeResult.O;
+            }
+            else if (finalBoard[2, 0] == 'O' && finalBoard[1, 1] == 'O' && finalBoard[0, 2] == 'O')
+            {
+                return TicTacToeResult.O;
+            }
+            else if(finalBoard[0, 0] == 'X' && finalBoard[1, 0] == 'X' && finalBoard[2, 0] == 'X')
+            {
+                return TicTacToeResult.X;
+            }
+            else if (finalBoard[0, 0] == 'X' && finalBoard[0, 1] == 'X' && finalBoard[0, 2] == 'X')
+            {
+                return TicTacToeResult.X;
+            }
+            else if (finalBoard[0, 0] == 'X' && finalBoard[1, 1] == 'X' && finalBoard[2, 2] == 'X')
+            {
+                return TicTacToeResult.X;
+            }
+            else if (finalBoard[1, 0] == 'X' && finalBoard[1, 1] == 'X' && finalBoard[1, 2] == 'X')
+            {
+                return TicTacToeResult.X;
+            }
+            else if (finalBoard[2, 0] == 'X' && finalBoard[2, 1] == 'X' && finalBoard[2, 2] == 'X')
+            {
+                return TicTacToeResult.X;
+            }
+            else if (finalBoard[2, 0] == 'X' && finalBoard[1, 1] == 'X' && finalBoard[0, 2] == 'X')
+            {
+                return TicTacToeResult.X;
+            }
+            else if (finalBoard[0, 1] == 'X' && finalBoard[1, 1] == 'X' && finalBoard[2, 1] == 'X')
+            {
+                return TicTacToeResult.X;
+            }
+            else if (finalBoard[0, 2] == 'X' && finalBoard[1, 2] == 'X' && finalBoard[2, 2] == 'X')
+            {
+                return TicTacToeResult.X;
+            }
+            else
+            {
+                return TicTacToeResult.Draw;
+            }
+        }
+        public bool EachArrayInJaggedArrayContainsTargetNumber(int[][] numbers, int targetNumber)
+        {
+            if (numbers.Length == 0)
+            { return false; }
+
+            int outerarrays = 0;
+            int contains = 0;
```

Comment:
This variable seems to be used like a boolean flag. Consider something like 'bool containsTarget = false;', then flip it to true on line 145. Line 148 would then be `if (containsTarget) { ... }`